src/idl_gen_cpp.cpp

```
@@ -569,26 +569,46 @@ class CppGenerator : public BaseGenerator {
                             : name;
   }
 
-  const std::string &PtrType(const FieldDef *field) {
-    auto attr = field ? field->attributes.Lookup("cpp_ptr_type") : nullptr;
-    return attr ? attr->constant : parser_.opts.cpp_object_api_pointer_type;
+  const std::string FieldStringLookup(const FieldDef *field,
+                                      const std::string &attrName) {
+    auto attr = field ? field->attributes.Lookup(attrName) : nullptr;
+    if (attr) {
+      return attr->constant;
+    } else {
+      if (attrName == "cpp_ptr_type") {
+        return parser_.opts.cpp_object_api_pointer_type;
+      } else if (attrName == "cpp_str_type") {
+        return parser_.opts.cpp_object_api_string_type;
+      } else if (attrName == "cpp_vec_type") {
+        return parser_.opts.cpp_object_api_vector_type;
+      } else {
+        return "";
```

can't silently return empty string, please assert.